linkatos/activities.py

```diff
@@ -1,45 +1,45 @@
-import time
 import linkatos.parser as parser
-import linkatos.confirmation as confirmation
 import linkatos.printer as printer
-import linkatos.utils as utils
 import linkatos.firebase as fb
+import linkatos.reaction as react
 
 
-def keep_wanted_urls(expecting_confirmation, url, slack_client, BOT_ID,
-                     fb_credentials, firebase):
-    READ_WEBSOCKET_DELAY = 1  # 1 second delay between reading from firehose
+def is_empty(events):
+    return ((events is None) or (len(events) == 0))
 
-    # parse the messages. Get a dictionary with @out, @channel,
-    # @out_type
-    parsed_message = parser.parse(slack_client.rtm_read(), BOT_ID)
 
-    print(parsed_message)
-    print(expecting_confirmation)
+def is_url(url_message):
+    return url_message['type'] == 'url'
 
-    if utils.is_fresh_url(expecting_confirmation, parsed_message['type']):
-        url = parsed_message['out']
 
-    print(url)
+def event_consumer(expecting_url, expecting_reaction, parsed_url_message,
+                   slack_client, fb_credentials, firebase):
 
-    printer.ask_confirmation(expecting_confirmation, parsed_message, slack_client)
+    # Read slack events
+    events = slack_client.rtm_read()
 
-    # update expecting_confirmation
-    # when it's a url
-    expecting_confirmation = confirmation.update_if_url(parsed_message,
-                                                        expecting_confirmation)
+    if is_empty(events):
+        return (expecting_url, expecting_reaction, parsed_url_message)
 
-    # check if there is an answer
-    (expecting_confirmation, is_yes) = confirmation.process_if_yn(parsed_message,
-                                                                  expecting_confirmation)
+    for event in events:
+        print(event)
+        print('expecting_url: ', expecting_url)
 
-    # printer.notify_confirmation(expecting_confirmation, is_yes)
+        if expecting_url and event['type'] == 'message':
+            parsed_url_message = parser.parse_url_message(event)
 
-    # Store url
-    if is_yes:
-        fb.connect_and_store_url(url, fb_credentials, firebase)
-        is_yes = False
+            if is_url(parsed_url_message):
+                printer.ask_confirmation(parsed_url_message, slack_client)
+                expecting_url = False
 
-    time.sleep(READ_WEBSOCKET_DELAY)
+        if not expecting_url and event['type'] == 'reaction_added':
```

Comment:
This means you no longer use `expecting_reaction` right? If so, we can remove it from everywhere.